homeassistant/components/device_tracker/__init__.py

```diff
@@ -396,9 +395,6 @@ def state_attributes(self):
             attr[ATTR_LONGITUDE] = self.gps[1]
             attr[ATTR_GPS_ACCURACY] = self.gps_accuracy
 
-        if self.battery:
-            attr[ATTR_BATTERY] = self.battery
-
         if self.attributes:
             for key, value in self.attributes.items():
```

Comment:
Actually skip my last comment and let's change this line to  attr.update(self.attributes)